cosipy/image_deconvolution/RichardsonLucy.py

```diff
@@ -193,21 +214,23 @@ def register_result(self):
         - loglikelihood: log-likelihood
         """
         
-        this_result = {"iteration": self.iteration_count, 
-                       "model": copy.deepcopy(self.model), 
-                       "delta_model": copy.deepcopy(self.delta_model),
-                       "processed_delta_model": copy.deepcopy(self.processed_delta_model),
-                       "background_normalization": copy.deepcopy(self.dict_bkg_norm),
-                       "alpha": self.alpha, 
-                       "loglikelihood": copy.deepcopy(self.loglikelihood_list)}
-
-        # show intermediate results
-        logger.info(f'  alpha: {this_result["alpha"]}')
-        logger.info(f'  background_normalization: {this_result["background_normalization"]}')
-        logger.info(f'  loglikelihood: {this_result["loglikelihood"]}')
+        # Master
+        if self.MASTER:
+            this_result = {"iteration": self.iteration_count, 
+                        "model": copy.deepcopy(self.model), 
+                        "delta_model": copy.deepcopy(self.delta_model),
+                        "processed_delta_model": copy.deepcopy(self.processed_delta_model),
+                        "background_normalization": copy.deepcopy(self.dict_bkg_norm),
+                        "alpha": self.alpha, 
+                        "loglikelihood": copy.deepcopy(self.loglikelihood_list)}
+
+            # show intermediate results
+            logger.info(f'  alpha: {this_result["alpha"]}')
+            logger.info(f'  background_normalization: {this_result["background_normalization"]}')
+            logger.info(f'  loglikelihood: {this_result["loglikelihood"]}')
```

Comment:
If you agree to add self.parallel_computation and self.master_node, I want to add the following lines at the beginning of this class function;
```
if self.parallel_computation == True and self.master_node == False:
    pass
```
I think that it is easier to understand what happens in the child nodes.

Reply:
Changes accepted.